### ISSUE TITLE
Investigate GRPC Tracer to support filtering or exclude the Health Checking Protocol.

### ISSUE THREAD

**Issue by marvinkite** [open-telemetry/opentelemetry-go-contrib#355]
If my GRPC service implements the [GRPC Health Checking Protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) the GRPC Client library and my Kubernetes health probe call it regularly and generates a "noise" only. I'd like to filter out them the same way I could filter in HTTP the Health check. It could also be part of the library as bool option to ignore these calls.

Something like:

```golang
if stream := grpc.ServerTransportStreamFromContext(ctx); stream != nil {
  if stream.Method() == "/grpc.health.v1.Health/Check" || stream.Method() == "/grpc.health.v1.Health/Watch" {
    //Ignore the trace
  }
}
```

See HTTP [Filters](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/master/instrumentation/net/http/config.go#L39)

```golang
// config represents the configuration options available for the http.Handler
// and http.Transport types.
type config struct {
	Filters           []Filter
}
```

**Comment by MrMegaMango** [comment]
is this implemented?

**Comment by dmathieu** [comment]
Yes, this is doable with the [WithFilter](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc#WithFilter) option.
